src/composed/signed_key/secret.rs

```diff
@@ -315,3 +315,23 @@ impl PublicKeyTrait for SignedSecretSubKey {
         self.key.to_writer_old(writer)
     }
 }
+
+impl From<SignedSecretKey> for SignedPublicKey {
+    fn from(value: SignedSecretKey) -> Self {
+        let public_primary = value.primary_key.public_key();
+        let details = value.details;
+
+        let mut public_subkeys = value.public_subkeys;
+
+        for subkey in value.secret_subkeys {
+            let pk = subkey.key.public_key();
+            let sig = subkey.signatures;
+
+            let spsk = SignedPublicSubKey::new(pk, sig);
```

Comment:
maybe the `From` should exist for subkeys as well? cleaning this code up a bit

eg 

```rust
let public_subkeys: Vec<SignedPublicSubKey> = value.secret_subkeys.into_iter().map(Into::into).collect();
```

Reply:
Good idea. Will do.

Reply:
I've adjusted the code (it's a bit less "neat", because two formerly separate sets of subkeys need to be joined together into one combined `Vec`).